tests/primitives/test_tool.py

```diff
@@ -159,3 +159,17 @@ def foo(x=100):
     tool = Tool(foo)
     assert tool.args["x"]["default"] == 100
     assert not hasattr(tool.args["x"], "type")
+
+
+def test_tool_call_parses_args():
+    tool = Tool(dummy_with_pydantic)
+
+    args = {
+        "model": {
+            "field1": "hello",
+            "field2": 123,
+        }
+    }
+
+    result = tool(**args)
```

Comment:
Q: What are the types we officially support as tool-callable for ReAct? Python primitives + Pydantic?

Reply:
yes, everything supported by typing + pydantic